perun/check/factory.py

```diff
@@ -109,7 +109,7 @@ def degradation_in_minor(minor_version, quiet=False):
     :returns: list of found changes
     """
     minor_version_info = vcs.get_minor_version_info(minor_version)
-    baseline_version_queue = minor_version_info.parents
+    baseline_version_queue = [p for p in minor_version_info.parents]
```

Comment:
Well, this was a very subtle bug, that didn't manifest during tests until I optimized the performance by caching chosen GIT/VCS functions. Since in python lists are mutable, the algorithm, that used the `baseline_version_queue` changed the `.parents` as well. So subsequent calls to GIT/VCS function returned minor version info with malformed `.parents` resulting into failed tests.